homework/index.js

```diff
@@ -30,18 +15,107 @@
+            class: 'contNumber',
+            text: contributor.contributions,
+          });
+        });
+      })
+      .catch(err => {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+      });
+  }
+
+  function fetchSingleRepoData(data, index, wrapper) {
+    const filteredData = data.filter(el => el.id === Number(index));
+
+    wrapper.innerHTML = '';
+    const repositoriesSection = createAndAppend('div', wrapper, { class: 'repInfo-side' });
+    const repositoryName = createAndAppend('div', repositoriesSection, { class: 'repName' });
+    const repositoryDescription = createAndAppend('div', repositoriesSection, {
+      class: 'repDescription',
+    });
+    const repositoryForks = createAndAppend('div', repositoriesSection, {
+      class: 'repForks',
+    });
+    const repositoryDate = createAndAppend('div', repositoriesSection, { class: 'repDate' });
+    filteredData.forEach(allData => {
+      createAndAppend('span', repositoryName, { text: 'Repository Name:' });
+      createAndAppend('a', repositoryName, {
+        text: allData.name,
+        href: allData.html_url,
+        target: '_blank',
+      });
+      createAndAppend('span', repositoryDescription, { text: 'Description:' });
+      createAndAppend('p', repositoryDescription, { text: allData.description });
+      createAndAppend('span', repositoryForks, { text: 'Forks:' });
+      createAndAppend('p', repositoryForks, { text: allData.forks });
+      createAndAppend('span', repositoryDate, { text: 'Date:' });
+      createAndAppend('p', repositoryDate, { text: allData.updated_at });
+      /* Fetch Contributor Data */
+      fetchContributorsData(allData, wrapper);
     });
   }
+  function fetchAllRepoData(url) {
+    return fetch(url)
+      .then(res => res.json())
+      .then(fetchedData => {
+        const navDiv = createAndAppend('div', root, { class: 'nav' });
+        const navHeader = createAndAppend('div', navDiv, { class: 'nav-header' });
+        const navTitle = createAndAppend('div', navHeader, { class: 'nav-title' });
+        navTitle.innerText = 'HYF Repositories';
+        const selectEl = createAndAppend('select', navDiv, { id: 'getRepoData' });
+        fetchedData.sort((a, b) => (a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1));
+        fetchedData.forEach(res => {
+          createAndAppend('option', selectEl, { text: res.name, value: res.id });
+        });
+        const selectRepo = document.getElementById('getRepoData');
+        const container = createAndAppend('div', root, { class: 'container' });
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+        const fetchData = () => {
+          fetchSingleRepoData(fetchedData, selectEl.value, container);
+        };
+
+        fetchSingleRepoData(fetchedData, Number('82560415'), container);
```

Comment:
Hi Ahmat - having the ID value hardcoded here is quite risky. What happens if the ID changes of the repo in the BE. The code would break in such a case, wouldn't it? 🙂 

What you can do here is pass `fetchedData[0].id` instead of the hardcoded value.